cli/roger_promote.py

```diff
@@ -0,0 +1,348 @@
+
+        :Raises:
+        :cli.roger_utils.RogerPromoteError:
+
+        :Return [bool]: True if successful, False if otherwise
+        """
+        # Get instance
+        if instance:
+            rp = instance
+        else:
+            rp = cls()
+
+        # Get Namespace obj
+        args = rp.arg_parse().parse_args()
+
+        # Set framework based on app config
+        rp._set_framework(args.config, args.app_name)
+
+        # Get repo name
+        repo = rp._config_resolver('repo', args.app_name, args.config)
+        if not repo:
+            raise RogerPromoteError('Repo not found')
+
+        # Clone the repo
+        rp._clone_repo(repo)
+
+        # Locate roger_push.py
+        roger_push = rp._roger_push_script()
+
+        app_data = rp._app_config.getAppData(
+            rp.config_dir,
+            args.config,
+            args.app_name
+        )
+        print()
+
+        image_refs = app_data['containers']
+        failed_images = []
+        for image_ref in image_refs:
+            template_path = rp._get_template_path(
+                image_ref,
+                rp.config_dir,
+                args,
+                args.app_name
+            )
+
+            full_image_name = rp._image_name(
+                args.from_env,
+                args.config,
+                template_path
+            )
+            image_name = full_image_name.split('/', 1)[1]
+
+            # Execute the script
+            cmd = [
+                roger_push,
+                '--env', args.to_env,
+                args.app_name, rp._temp_dir, image_name, args.config
+            ]
+            string_cmd = ' '.join(cmd)
+
+            ret_val = None
+            x = 0
+            while x < 3:
+                x += 1
+                ret_val = os.system(string_cmd)
+                if ret_val == 0:
+                    break
+
+            if ret_val != 0:
+                print("Roger failed to deploy {image} to {env}".format(
+                    image_name, args.to
+                ))
+                failed_images.append(image_name)
+
+        # CleanUp
+        shutil.rmtree(rp._temp_dir)
+        if len(failed_images) > 0:
+            print("Images that failed")
+            for failed_image in failed_images:
+                print(failed_image)
+            return False
+        return True
+
+    def arg_parse(self):
+        """
+        Returns a argparse.NameSpace instance holding argument data
+        """
+        parser = argparse.ArgumentParser(
+            prog='roger promote',
+            description=describe()
+        )
+
+        env_choices = ['local', 'dev', 'stage', 'prod']
+
+        parser.add_argument(
+            'from_env', choices=env_choices, help='The source environment'
+        )
+        parser.add_argument(
+            'to_env', choices=env_choices, help='The destination environment'
+        )
+        parser.add_argument('config', help='The name of the config file')
```

Comment:
The app name should come before the config file to remain consistent with the push interface.